openmrs/api/src/main/java/org/projectbuendia/openmrs/api/impl/ProjectBuendiaServiceImpl.java

```diff
@@ -13,23 +13,28 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.openmrs.Encounter;
 import org.openmrs.api.impl.BaseOpenmrsService;
 import org.projectbuendia.openmrs.api.ProjectBuendiaService;
 import org.projectbuendia.openmrs.api.db.ProjectBuendiaDAO;
 
+import javax.annotation.Nullable;
+import java.util.Date;
+import java.util.List;
+
 /** It is a default implementation of {@link ProjectBuendiaService}. */
 public class ProjectBuendiaServiceImpl extends BaseOpenmrsService implements ProjectBuendiaService {
     protected final Log log = LogFactory.getLog(this.getClass());
 
     private ProjectBuendiaDAO dao;
 
-    /** @return the dao */
-    public ProjectBuendiaDAO getDao() {
-        return dao;
+    @Override
+    public void setDAO(ProjectBuendiaDAO dao) {
+        this.dao = dao;
     }
 
-    /** @param dao the dao to set */
-    public void setDao(ProjectBuendiaDAO dao) {
-        this.dao = dao;
+    @Override
+    public List<Encounter> getEncountersModifiedOnOrAfter(@Nullable Date date) {
```

Comment:
Just wondering -- is there any way to make this return an iterator, so that if we want to return just the first page of 100 results we can do that without fetching the entire list?  Or are we stuck with the whole list because Criteria offers no other option?